pkg/recommend/advisor/ehpa.go

```diff
@@ -0,0 +1,170 @@
+package advisor
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	autoscalingv2 "k8s.io/api/autoscaling/v2beta2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	analysisapi "github.com/gocrane/api/analysis/v1alpha1"
+	autoscalingapi "github.com/gocrane/api/autoscaling/v1alpha1"
+	predictionapi "github.com/gocrane/api/prediction/v1alpha1"
+
+	"github.com/gocrane/crane/pkg/prediction/config"
+	"github.com/gocrane/crane/pkg/recommend/types"
+	"github.com/gocrane/crane/pkg/utils"
+)
+
+var _ Advisor = &EHPAAdvisor{}
+
+type EHPAAdvisor struct {
+	*types.Context
+}
+
+func (a *EHPAAdvisor) Advise(proposed *types.ProposedRecommendation) error {
+	p := a.Predictors[predictionapi.AlgorithmTypePercentile]
+
+	mc := &config.MetricContext{}
+
+	resourceCpu := corev1.ResourceCPU
+	namespace := a.Recommendation.Spec.TargetRef.Namespace
+	if len(namespace) == 0 {
+		namespace = DefaultNamespace
+	}
+	cpuQueryExpr := ResourceToPromQueryExpr(namespace, a.Recommendation.Spec.TargetRef.Name, &resourceCpu)
+
+	klog.V(4).Infof("EHPAAdvisor CpuQuery %s Recommendation %s", cpuQueryExpr, klog.KObj(a.Recommendation))
+	timeNow := time.Now()
+	tsList, err := a.DataSource.QueryTimeSeries(cpuQueryExpr, timeNow.Add(-time.Hour*24*7), timeNow, time.Minute)
```

Comment:
Why need historical metrics?

Reply:
When recommend minReplicas and maxReplicas, maybe consider both the history and future is more reliable.